cli/openapi3_test.go

```diff
@@ -0,0 +1,498 @@
+				Required:    true,
+				Description: "peach resource input",
+				Content: openapi3.Content{
+					"application/json": &openapi3.MediaType{
+						Schema: &openapi3.SchemaRef{Value: &openapi3.Schema{
+							Type: "object",
+							Properties: map[string]*openapi3.SchemaRef{
+								"sweetness": {Value: &openapi3.Schema{
+									Description: "Peach sweetness description",
+									Type:        "string",
+									ExtensionProps: openapi3.ExtensionProps{Extensions: map[string]interface{}{
+										"title": "Peach sweetness title",
+									}},
+								}},
+							},
+							AdditionalPropertiesAllowed: boolean(false),
+						}},
+					},
+				},
+			}}
+
+			response := root.Paths["/peaches/{id}"].Put.RequestBody
+			Expect(response).To(WithTransform(transformJSON, Equal(transformJSON(expectedResponse))))
+		})
+	})
+
+	It("operations should contains tags based on resource_group", func() {
+		multiple := root.Paths["/shops"]
+		single := root.Paths["/shops/{id}"]
+		operations := []*openapi3.Operation{
+			multiple.Get,
+			multiple.Post,
+			single.Get,
+			single.Put,
+			single.Delete,
+		}
+
+		expectedTags := []string{"Resource Group"}
+		for _, operation := range operations {
+			Expect(operation.Tags).To(Equal(expectedTags))
+		}
+	})
+})
+
+func removeRef(i interface{}) {
+	removeRefRecursive(reflect.ValueOf(i))
+}
+
+func removeRefRecursive(value reflect.Value) {
+	if !value.IsValid() {
+		return
+	}
+	switch value.Kind() {
+	case reflect.Ptr:
+		removeRefRecursive(value.Elem())
+
+	case reflect.Struct:
+		for i := 0; i < value.NumField(); i += 1 {
+			if value.Type().Field(i).Name == "Ref" {
+				value.Field(i).SetString("")
+			}
+			removeRefRecursive(value.Field(i))
+		}
+	case reflect.Slice:
+		for i := 0; i < value.Len(); i += 1 {
+			removeRefRecursive(value.Index(i))
+		}
+	case reflect.Map:
+		for _, key := range value.MapKeys() {
+			removeRefRecursive(value.MapIndex(key))
+		}
+	}
+}
+
+func transformJSON(value interface{}) interface{} {
+	// wrapping in some other object, allows arbitrary object to be marshaled and unmarshaled
+	buf, err := json.Marshal([]interface{}{value})
+	if err != nil {
+		panic(err)
+	}
+	result := []interface{}{}
+	err = json.Unmarshal(buf, &result)
+	if err != nil {
+		panic(err)
+	}
+	return result[0]
+}
+
+func command(cmd cli.Command, args ...string) []byte {
+	app := cli.NewApp()
+	app.Commands = []cli.Command{cmd}
+	output, err := captureStdout(func() {
+		Expect(app.Run(append([]string{"gohan", cmd.Name}, args...))).To(Succeed())
+	})
+	Expect(err).NotTo(HaveOccurred())
+	return output
+}
+
+func captureStdout(f func()) ([]byte, error) {
+	stdout := os.Stdout
+	r, w, err := os.Pipe()
+	if err != nil {
+		return nil, err
+	}
+	Expect(err).NotTo(HaveOccurred())
+	os.Stdout = w
+	go func() {
+		f()
+		_ = w.Close()
+	}()
+	content, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+	os.Stdout = stdout
```

Comment:
Good idea, done.

Actually in case of error in this place, test would hang indefinitely (well, at least until ginkgo decides to kill it) because writer part of the pipe was never closed, and a reader was blocked. I fixed that as well.